elements/lisk-transactions/test/sign.spec.ts

```diff
@@ -0,0 +1,387 @@
+		];
+		return invalidAssets.forEach(transactionObject =>
+			expect(() =>
+				signMultiSignatureTransaction(
+					validAssetSchema,
+					transactionObject,
+					networkIdentifier,
+					passphrase1,
+					keys,
+				),
+			).toThrow(),
+		);
+	});
+
+	it('should sign and include sender signature at position zero for given sender passphrase', () => {
+		const senderTransaction = signMultiSignatureTransaction(
+			validAssetSchema,
+			{ ...validTransaction, signatures: [] },
+			networkIdentifier,
+			passphrase1,
+			keys,
+			true,
+		);
+
+		expect(senderTransaction.signatures).toStrictEqual([
+			getSignature(passphrase1, senderTransaction),
+			Buffer.alloc(0),
+			Buffer.alloc(0),
+			Buffer.alloc(0),
+			Buffer.alloc(0),
+		]);
+		expect(senderTransaction.signatures).toHaveLength(5);
+		return expect(senderTransaction).toMatchSnapshot();
+	});
+
+	it('should assign empty string for signatures of missing passphrases', () => {
+		const signedTransaction = signMultiSignatureTransaction(
+			validAssetSchema,
+			{ ...validTransaction, signatures: [] },
+			networkIdentifier,
+			passphrase2,
+			keys,
+			true,
+		);
+		const partialSignedTransaction = signMultiSignatureTransaction(
+			validAssetSchema,
+			{ ...signedTransaction, signatures: [...(signedTransaction.signatures as Array<Buffer>)] },
+			networkIdentifier,
+			passphrase3,
+			keys,
+			true,
+		);
+
+		expect(signedTransaction.signatures).toHaveLength(5);
+		expect(partialSignedTransaction.signatures).toHaveLength(5);
+		expect(signedTransaction.signatures).toStrictEqual([
+			Buffer.alloc(0),
+			getSignature(passphrase2, signedTransaction),
+			Buffer.alloc(0),
+			Buffer.alloc(0),
+			Buffer.alloc(0),
+		]);
+		expect(partialSignedTransaction.signatures).toStrictEqual([
+			Buffer.alloc(0),
+			getSignature(passphrase2, signedTransaction),
+			Buffer.alloc(0),
+			getSignature(passphrase3, partialSignedTransaction),
+			Buffer.alloc(0),
+		]);
+		expect(signedTransaction).toMatchSnapshot();
+		return expect(partialSignedTransaction).toMatchSnapshot();
+	});
+
+	it('should sort keys and sign transaction', () => {
+		const unSortedKeys = {
+			mandatoryKeys: [publicKey1, publicKey2],
+			optionalKeys: [publicKey4, publicKey3],
+		};
+		const transactionObject = { ...validTransaction, signatures: [] };
+		const unSignedTransaction = { ...validTransaction, signatures: [] };
+
+		[passphrase1, passphrase2, passphrase3, passphrase4].forEach(passphrase =>
+			signMultiSignatureTransaction(
+				validAssetSchema,
+				transactionObject,
+				networkIdentifier,
+				passphrase,
+				unSortedKeys,
+				true,
+			),
+		);
+
+		expect(transactionObject.signatures).toStrictEqual([
+			getSignature(passphrase1, unSignedTransaction),
+			getSignature(passphrase2, unSignedTransaction),
+			Buffer.alloc(0),
+			getSignature(passphrase3, unSignedTransaction),
+			getSignature(passphrase4, unSignedTransaction),
+		]);
+		return expect(transactionObject).toMatchSnapshot();
+	});
+});
```

Comment:
maybe it would be nice to have a test that takes the valid register transaction without signatures, and sign with all the passphrase, then check if the passphrases is correct using the protocol-spec? (for includeSender true/false)